### PR TITLE
Include H(message_1) in Voucher

### PR DESCRIPTION
**Pull request by gselander**
Fixing Editor's note about H(message_1) in Voucher

### CODE REVIEW

**Comment by malishav** [comment]
Would it be safe to rely completely on H(message_1) instead of G_X as a nonce in the protocol? This would simplify the implementation in a sense that we would no longer need to access private elements of the EDHOC implementation, given that message_1 is public info that needs to be transported over the wire anyways and we can hash it (again) outside of the EDHOC implementation.

**Comment by gselander** [comment]
H(message_1) can serve as unique session identifier, and replace G_X in VRES. But G_X is still needed in VREQ for W to build the shared secret G_XW.

You mention "private", but G_X is the public key and sent in message_1 so it is not so private . . .

**Comment by malishav** [comment]
Yes ofc, shared secret.. I meant private in a sense of an EDHOC API that will typically not expose this. Parsing message_1 to get to `G_X` would work but is dirty.

**Comment by gselander** [comment]
@malishav I made H(m1) session id - have a look.

File: draft-selander-ace-ake-authz.md [hunk 1]
```diff
@@ -327,19 +328,15 @@ where context is a CBOR bstr wrapping of the following CBOR sequence:
 ~~~~~~~~~~~
 voucher_input = (
     V_TYPE:        int,
-    SS:            int,
-    G_X:           bstr,
-    ID_U:          bstr,
+    H(message_1)   bstr,
```
Comment (malishav) [inline draft-selander-ace-ake-authz.md:331]: missing colon

File: draft-selander-ace-ake-authz.md [hunk 2]
```diff
@@ -423,6 +418,7 @@ V sends the voucher request to W. The Voucher Request SHALL be a CBOR array as d
 
 ~~~~~~~~~~~
 Voucher_Request = [
+    H(message_1)     bstr,
```
Comment (malishav) [inline draft-selander-ace-ake-authz.md:421]: missing colon

**Review by malishav — APPROVED** [review]

**Pull request merged**